### ISSUE TITLE
100% CPU utilization of AWS EC2 Worker Nodes 

### ISSUE THREAD

**Issue by robertsmol** [edgelesssys/constellation#3383]
### Issue description

I am attempting to deploy Kubernetes pods in a constellation cluster in AWS. I noticed that my pods were struggling to start successfully and cycling through a number or restarts. When the pods finally did start they were unresponsive. I was able to establish a connection to them but unable to get a response to my request.

I took a look at the individual worker nodes in AWS and all workers were running at 100% CPU utilization. I terminated the cluster and deployed a new one without deploying any of my components. The workers  seem to run at 58% until the constellation component are installed and the  CPU utilization then goes up to 100%. This is consistent on all 3 of my worker nodes and the CPU utilization does not drop off.

### Steps to reproduce the behavior

Deploy a new constellation cluster in AWS. Monitor the CPU utilization of the Worker Nodes before, during and after the installation of the constellation components

### Version

v2.17.0, v2.18.0

### Constellation Config

``` yaml
version: v4 # Schema version of this configuration file.
image: v2.18.0 # Machine image version used to create Constellation nodes.
name: constell # Name of the cluster.
kubernetesVersion: v1.29.8 # Kubernetes version to be installed into the cluster.
microserviceVersion: v2.18.0 # Microservice version to be installed into the cluster. Defaults to the version of the CLI.
debugCluster: false # DON'T USE IN PRODUCTION: enable debug mode and use debug images.
customEndpoint: "" # Optional custom endpoint (DNS name) for the Constellation API server.
internalLoadBalancer: false # Flag to enable/disable the internal load balancer. If enabled, the Constellation is only accessible fro.
serviceCIDR: 10.96.0.0/12 # The Kubernetes Service CIDR to be used for the cluster. This value will only be used during the first ini.
# Additional tags that are applied to created resources.
tags: {}
# Supported cloud providers and their specific configurations.
provider:
    # Configuration for AWS as provider.
    aws:
        region: "us-east-2" # AWS data center region. See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availabis
        zone: "us-east-2a" # AWS data center zone name in defined region. See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/uss
        iamProfileControlPlane: "constellsdps_control_plane_instance_profile" # Name of the IAM profile to use for the control-plane .
        iamProfileWorkerNodes: "constellsdps_worker_node_instance_profile" # Name of the IAM profile to use for the worker nodes.
        deployCSIDriver: true # Deploy Persistent Disk CSI driver with on-node encryption. For details see: https://docs.edgeless.syse
        useMarketplaceImage: false # Use the specified AWS Marketplace image offering.
# Node groups to be created in the cluster.
nodeGroups:
    control_plane_default:
        role: control-plane # Role of the nodes in this group. Valid values are "control-plane" and "worker".
        zone: "us-east-2a" # Availability zone to place the VMs in.
        instanceType: m6a.xlarge # VM instance type to use for the nodes.
        stateDiskSizeGB: 30 # Size (in GB) of a node's disk to store the non-volatile state.
        stateDiskType: gp3 # Type of a node's state disk. The type influences boot time and I/O performance.
        initialCount: 3 # Number of nodes to be initially created.
    worker_default:
        role: worker # Role of the nodes in this group. Valid values are "control-plane" and "worker".
        zone: "us-east-2a" # Availability zone to place the VMs in.
        instanceType: m6a.xlarge # VM instance type to use for the nodes.
        stateDiskSizeGB: 30 # Size (in GB) of a node's disk to store the non-volatile state.
        stateDiskType: gp3 # Type of a node's state disk. The type influences boot time and I/O performance.
        initialCount: 3 # Number of nodes to be initially created.
```

**Comment by 3u13r** [comment]
Thanks for the report. It seems like only AWS with SEV-SNP is affected. I see >60% CPU utilization after all components are up. 

I verified that the issue is (1) not a general issue since AWS without SNP and (2) SNP on GCP both show reasonable CPU utilization numbers.

So while we investigate this issue further, we recommend either switching to another CSP or, if you currently just trying out Constellation and deploying nothing sensitive into it, then switching to `aws-nitro-tpm` attestation.

**Comment by robertsmol** [comment]
Thanks. I can confirm that this is also not an issue in Azure.

**Comment by burgerdev** [comment]
Hi @robertsmol,

We found the issue, it's fixed with #3387 and should be part of release v2.19.0, which is supposed to happen in the next week.

As far as I understand the problem, it's not quite as bad as the monitoring made it look. Under low load, the Kernel was configured to poll for new work in a tight loop instead of halting the CPU. When observed from the outside this might be hard to distinguish from the CPU doing actual work, at least that's how I interpret the 100% utilization metrics. Once the kernel has something meaningful to do, the polling should go down.

However, there's still a performance overhead with polling, energy is being wasted, and the metrics are basically unusable. Fortunately, the issue why we introduced `idle=poll` in the first place was fixed at AWS, so we are now removing it.

Thanks for reporting the bug!

Cheers, Markus
